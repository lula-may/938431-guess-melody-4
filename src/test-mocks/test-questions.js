const questions = [
  {
    type: `singer`,
    song: {
      singer: `Lady Gaga`,
      src: `https://music/Lady_Gaga.ogg`
    },
    answers: [
      {
        singer: `Lady Gaga`,
        avatar: `https://images/Lady_Gaga.jpg`,
        id: `0`
      },
      {
        singer: `Witney Houston`,
        avatar: `https://images/Witnew_Houston.jpg`,
        id: `1`
      },
      {
        singer: `Cher`,
        avatar: `https://images/Cher.jpg`,
        id: `2`
      }
    ]
  },
  {
    type: `genre`,
    genre: `jazz`,
    answers: [
      {
        src: `https://samples/1.ogg`,
        genre: `blues`,
        id: `3`
      },
      {
        src: `https://samples/2.ogg`,
        genre: `jazz`,
        id: `4`
      },
      {
        src: `https://samples/3.ogg`,
        genre: `folk`,
        id: `5`
      },
      {
        src: `https://samples/4.ogg`,
        genre: `jazz`,
        id: `6`
      }
    ]
  }
];

export {questions};
