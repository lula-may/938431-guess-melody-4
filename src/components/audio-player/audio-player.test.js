import React from "react";
import renderer from "react-test-renderer";
import AudioPlayer from "./audio-player.jsx";
import {questions} from "../../test-mocks/test-questions";

const mockSrc = questions[0].song.src;

it(`Should correctly render AudioPlayer Component`, () => {
  const tree = renderer.create(
      <AudioPlayer
        src={mockSrc}
        isPlaying={false}
      />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
