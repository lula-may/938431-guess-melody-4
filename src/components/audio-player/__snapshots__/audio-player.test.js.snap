// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should correctly render AudioPlayer Component 1`] = `
Array [
  <button
    className="track__button track__button--play"
    disabled={true}
    onClick={[Function]}
    type="button"
  />,
  <div
    className="track__status"
  >
    <audio />
  </div>,
]
`;
